### PR TITLE
P2-948 - Make the Schema analysis work with blocks

### PR DESCRIPTION
**Pull request by hansjovis**
## Summary

This PR can be summarized in the following changelog entry:

* [@yoast/schema-blocks] Makes the Schema analysis work with the inner blocks now being placed directly within a post, instead of within an overarching Schema block.

## Relevant technical choices:

* 

## Test instructions

This PR can be tested by following these steps:

* Companion PR of https://github.com/Yoast/wordpress-seo/pull/17022. Please test this once using the test instructions as described in that PR.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Quality assurance

* [ ] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes #

### CODE REVIEW

File: packages/schema-blocks/src/functions/gutenberg/watch.ts [hunk 1]
```diff
@@ -145,7 +145,14 @@ export default function watch() {
 
 			updatingSchema = true;
 			{
-				const validations: BlockValidationResult[] = validateBlocks( rootBlocks );
+				const validationsForExistingBlocks = validateBlocks( rootBlocks );
```
Comment (johannadevos) [inline packages/schema-blocks/src/functions/gutenberg/watch.ts:148]: Where did the type go?

**Comment by johannadevos** [comment]
CR: ✅ 

- [x] `innerBlocksSidebar` has been renamed to `SchemaAnalysis`.
- [x] In `SchemaAnalysis`, there are no occurrences of the strings `parent` or `clientId`.
- [x] The selectors `getRequiredBlockNames` and `getRecommendedBlockNames` return a hard-coded array of block names and these arrays are complete.
- [x] `getValidationsForBlockNames` (previously `getInnerblockValidations`) has been rewritten to no longer use the parent `clientId` and directly return the array of `BlockValidationResult`s for the job posting blocks.
- [x] The sidebar is no longer rendered in the `innerBlocks` instruction.

I do see a couple of issues which will need to be split out to new tickets:
- [ ] There are a lot of traces left in the monorepo of the old Job Posting block, for example the `inheritSidebar` instruction. These will need to be removed. Many functions might need to be renamed (for example `sidebarWarningPresenter` - maybe we can just call it `warningPresenter` and leave the location unspecified).
- [ ] `useValidationResults` in `SchemaAnalysis.tsx` passes all the block names from the Yoast SEO Schema store to the `getValidationsForBlockNames` selector. This will become a problem when we have multiple structured data blocks (the analyses will become mixed up). The Schema analysis should apply to each of the (structured data blocks-related) graph pieces in turn. But I think we best leave this until the moment we add a new structured data block. There are some UX questions about this.

File: packages/schema-blocks/src/functions/gutenberg/watch.ts (hunk 1 above)
Comment (hansjovis) [reply packages/schema-blocks/src/functions/gutenberg/watch.ts:148]: TypeScript finds out automatically that `validateBlocks` returns an array of `BlockValidationResult` from the function's signature (e.g. `function validateBlocks( blocks: BlockInstance[] ): BlockValidationResult[]`). So a type is not strictly necessary.

Comment (johannadevos) [reply packages/schema-blocks/src/functions/gutenberg/watch.ts:148]: Ah 💡

**Comment by johannadevos** [comment]
Merging despite the failing build, because the failing unit test is in another package (`yoastseo`) and unrelated to our PR.

**Pull request merged**